### ISSUE TITLE
ProbabilityDensityAnalysis results in strange chargcar 

### ISSUE THREAD

**Issue by thienbinh92** [materialsvirtuallab/pymatgen-analysis-diffusion#310]
Hello,
I get a strange result from ProbabilityDensityAnalysis. Li and transition metal layers are not arranged like POSCAR/CONTCAR. Here is the code I use:
`file = ["tem-1000/out_vasprun.xml"]`
`analyzer = DiffusionAnalyzer.from_files(file, specie="Li", smoothed=False)`
`structure = analyzer.structure`
`trajectories = [s.frac_coords for s in analyzer.get_drift_corrected_structures()]`
`pda = ProbabilityDensityAnalysis(structure, trajectories, species="Li")`
`pda.to_chgcar("CHGCAR.vasp")`
I attach vasprun file below. Hope that someone help me to check.

<img width="455" alt="Screenshot 0004-05-19 at 0 24 26" src="https://user-images.githubusercontent.com/56110825/169081611-3ae07808-ff43-447e-bc5c-2eb0c65f807a.png">
[AIMD.zip](https://github.com/materialsvirtuallab/pymatgen-analysis-diffusion/files/8717660/AIMD.zip)

**Comment by JiQi535** [comment]
@thienbinh92 Can you explain more why you say "Li and transition metal layers are not arranged like POSCAR/CONTCAR"?

From my check, the CHGCAR.vasp from ProbabilityDensityAnalysis has the same atom coordinates as that of the initial structure in the AIMD snapshots you attached. I added `structure.to("cif","AIMD_01.cif")` to your code and compared AIMD_01.cif and CHGCAR.vasp.

**Comment by thienbinh92** [comment]
@JiQi535 Do you get "AIMD_01.cif" from vasprun.xml? From the CONTCAR file renamed to "out_cont.vasp", this material has an arrangement along c direction like this 3(-Li-O-TM-O-) (3 Li layers, 3 TM layers, and 6 O layers). When I open CHARGCAR by VESTA as shown in the picture above, one more TM layer is added. The arrangement has changed: Li layers are replaced by TM layers while TM layers become Li layers (compare "out_cont.vasp" and generated "CHARGCAR"). Because Li can diffuse, I think charge density (yellow color) should be around the Li atoms, however, it is around the TM atoms from CHARGCAR.

**Comment by JiQi535** [comment]
@thienbinh92 Yes, I got the AIMD_01.cif from vasprun.xml. The structure it contains should be equivalent to the initial structure of AIMD, instead of structure in CONTCAR.

With the code you provided, I generated one CHAGCAR.vasp file that does not have the problem you described.
[CHGCAR.vasp.zip](https://github.com/materialsvirtuallab/pymatgen-analysis-diffusion/files/8725350/CHGCAR.vasp.zip)

I believe what you are reporting is related to [a solved issue.](https://github.com/materialsvirtuallab/pymatgen-analysis-diffusion/issues/125) Please update your package with `git pull` and `pip install .`, then the issue should be solved.

**Comment by thienbinh92** [comment]
@JiQi535 I am not clear on how to upgrade. So I have just uninstalled pymatgen-analysis-diffusion and re-install with the command `pip install pymatgen-analysis-diffusion`. It still works incorrectly.

**Comment by thienbinh92** [comment]
@JiQi535 Thanks a lot. I downloaded and its result looks good.
PS: `git pull` does not work on my laptop. but `git clone https://github.com/materialsvirtuallab/pymatgen-analysis-diffusion` works well.
